### ISSUE TITLE
Add more diagnostics

### ISSUE THREAD

**Issue by forsyth2** [E3SM-Project/zppy#14]
After #2 is merged, we still need to add diurnal cycle, streamflow diagnostics, and ARM Diagnostics. These should be done as separate pull requests.

**Comment by forsyth2** [comment]
Streamflow diagnostics will likely require at least the following:

In `templates/default.ini`:
- Under `[ts]`: `var_xtr = string(default="")`
- Under the `[[__many__]]` subsection: `var_xtr = string(default=None)`
- Add `'streamflow'` to `sets` under `[e3sm_diags]` and `[e3sm_diags_vs_model]`

In `templates/e3sm_diags.bash`:
- Will likely need to generalize something like the following:
```
rofDir=/lcrc/group/e3sm/ac.forsyth2/E3SM_simulations/20210122.v2_test01.piControl.ne30pg2_EC30to60E2r2-1900_ICG.chrysalis/post/rof/native/ts/monthly/10yr
mkdir -p rof_links
cd rof_links
v="RIVER_DISCHARGE_OVER_LAND_LIQ"
xml_name=${v}_${Y1}01_${Y2}12.xml
cdscan -x ${xml_name} ${rofDir}/${v}_*.nc
cd ..
```

In the `e3sm.py` section of `templates/e3sm_diags.bash`:
- Add `from acme_diags.parameter.streamflow_parameter import StreamflowParameter`
- Add the streamflow parameter:
```
streamflow_param = StreamflowParameter()
streamflow_param.reference_data_path = '{{ obs_ts }}'
streamflow_param.test_data_path = test_ts
streamflow_param.test_name = short_name
streamflow_param.test_start_yr = start_yr
streamflow_param.test_end_yr = end_yr - # Streamflow gauge station data range from year 1986 to 1995
streamflow_param.ref_start_yr = "1986"
streamflow_param.ref_end_yr = "1995"
```
- Add `streamflow_param` to the `runner.run_diags` line

In `templates/ts.bash`:
- Add the following to the `ncclimo` call:
```
 {%- if var_xtr != '' %}
--var_xtr={{ var_xtr }} \
{%- endif %}
```
In the configuration file (not part of `zppy`):
- Add the following subsection under `[ts]`:
```
  [[ rof_monthly ]]
  input_subdir = "archive/rof/hist"
  input_files = "mosart.h0"
  frequency = "monthly"
  mapping_file = ""
  vars = "RIVER_DISCHARGE_OVER_LAND_LIQ"
  var_xtr = 'areatotal2'
```
- Add `RIVER_DISCHARGE_OVER_LAND_LIQ` to `[e3sm_diags]` > `[[ atm_monthly_180x360_aave ]]` >  `vars`

**Comment by forsyth2** [comment]
https://github.com/E3SM-Project/e3sm_diags/blob/master/examples/tutorials/postprocessing_E3SM_data_for_e3sm_diags.sh may be useful for adding diurnal cycle diagnostics.

**Comment by forsyth2** [comment]
Closing in favor of #31, #32, #33.
